r2i/preprocessing/normalize.cc

```diff
@@ -0,0 +1,164 @@
+    return error;
+  }
+
+  required_width = out_frame->GetWidth();
+  required_height = out_frame->GetHeight();
+  required_format = out_frame->GetFormat();
+  required_format_id = required_format.GetId();
+  required_channels = required_format.GetNumPlanes();
+
+  error = Validate(required_width, required_height, required_format_id);
+  if (error.IsError ()) {
+    return error;
+  }
+
+  PreProcessImage(in_data, out_data, width, height, channels, required_width,
+                  required_height, required_channels, error);
+
+  if (error.IsError ()) {
+    return error;
+  }
+  return error;
+}
+
+std::vector<r2i::ImageFormat> Normalize::GetAvailableFormats() {
+  return this->formats;
+}
+
+std::vector<std::tuple<int, int>>
+Normalize::GetAvailableDataSizes() {
+  return this->dimensions;
+}
+
+r2i::RuntimeError Normalize::Validate (int required_width,
+                                       int required_height,
+                                       r2i::ImageFormat::Id required_format_id) {
+
+  r2i::RuntimeError error;
+  r2i::ImageFormat format;
+  bool match_dimensions = false;
+  bool match_format = false;
+  int width = 0;
+  int height = 0;
+
+  /* Verify if the required dimensions are supported */
+  for (unsigned int i = 0; i < this->dimensions.size(); i++) {
+    width = std::get<0>(this->dimensions.at(i));
+    height = std::get<1>(this->dimensions.at(i));
+    if (width == required_width and height == required_height) {
+      match_dimensions = true;
+      break;
+    }
+  }
+
+  if (!match_dimensions) {
+    error.Set (r2i::RuntimeError::Code::MODULE_ERROR,
+               "Required output dimensions are not supported in the preprocessing module");
+    return error;
+  }
+
+  /* Verify if the required format is supported */
+  for (unsigned int i = 0; i < this->formats.size(); i++) {
+    format = this->formats.at(i);
+    if (format.GetId() == required_format_id) {
+      match_format = true;
+      break;
+    }
+  }
+
+  if (!match_format) {
+    error.Set (r2i::RuntimeError::Code::MODULE_ERROR,
+               "Required output image format is not supported in the preprocessing module");
+    return error;
+  }
+
+  return error;
+}
+
+r2i::RuntimeError Normalize::SetNormalizationParameters (
+  unsigned char *frame_data, int width, int height,
+  int channels) {
+  return r2i::RuntimeError();
+}
+
+void Normalize::PreProcessImage (unsigned char *in_data, float *out_data,
+                                 int width, int height, int channels, int required_width,
+                                 int required_height, int required_channels,
+                                 r2i::RuntimeError error) {
+
+  const int scaled_size = required_channels * required_width * required_height;
+
+  /* To set model specific preprocessing paramaters */
+  SetNormalizationParameters(in_data, required_width, required_height,
+                             channels);
+
+  for (int i = 0; i < scaled_size; i += channels) {
```

Comment:
What happened with all the logic that allowed RGB, RGBx, RGBA, BGR, BGRx, etc...?